logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/GetUserInfoUseCase.kt

```diff
@@ -19,22 +23,49 @@ fun interface GetUserInfoUseCase {
     suspend operator fun invoke(userId: UserId): GetUserInfoResult
 }
 
-internal class GetUserInfoUseCaseImpl(private val userRepository: UserRepository) : GetUserInfoUseCase {
+internal class GetUserInfoUseCaseImpl(
+    private val userRepository: UserRepository,
+    private val teamRepository: TeamRepository
+) : GetUserInfoUseCase {
 
     override suspend fun invoke(userId: UserId): GetUserInfoResult {
-        val otherUser = userRepository.getKnownUser(userId).firstOrNull()
-        if (otherUser != null) return GetUserInfoResult.Success(otherUser)
-        return userRepository.fetchUserInfo(userId)
-            .fold({
-                GetUserInfoResult.Failure
-            }, {
-                GetUserInfoResult.Success(it)
-            })
+        return getOtherUser(userId).fold(
+            { GetUserInfoResult.Failure },
+            { otherUser ->
+                getOtherUserTeam(otherUser).fold(
+                    { GetUserInfoResult.Failure },
+                    { team -> GetUserInfoResult.Success(otherUser, team) })
+            }
+        )
+    }
+
+    private suspend fun getOtherUser(userId: UserId): Either<CoreFailure, OtherUser> {
+        val localOtherUser = userRepository.getKnownUser(userId).firstOrNull()
```

Comment:
I know get DB operation are less meant to fail but still, shouldn't we wrap this repository call with `wrapStorageRequest` to make sure no error will be thrown? Otherwise if these operations fail, it can crash the client. It should be worth also to be added to the tests suite.

Reply:
done